core/src/main/java/org/geysermc/geyser/scoreboard/Objective.java

```diff
@@ -128,6 +137,21 @@ public Objective setDisplayName(String displayName) {
         return this;
     }
 
+    public Objective setNumberFormat(NumberFormat numberFormat) {
+        // Update the number format for scores that are following this objective's number format
+        for (Score score : scores.values()) {
+            if (score.getNumberFormat() == this.numberFormat) {
```

Comment:
The idea here was to check if the same object reference was used (that would mean it was set previously by this method or by `Objective#setScore`, and should follow the objective's number format). To better reflect how number formats are stored, I've changed the code to take the objective's number format whenever the score's number format is `null` instead of always having the score's number format set. See:
https://github.com/GeyserMC/Geyser/blob/ec22fda65f29d011d5b4d8892b9e26560d7d79d6/core/src/main/java/org/geysermc/geyser/scoreboard/Score.java#L173-L176
and
https://github.com/GeyserMC/Geyser/blob/ec22fda65f29d011d5b4d8892b9e26560d7d79d6/core/src/main/java/org/geysermc/geyser/entity/type/player/PlayerEntity.java#L430-L433